### PR TITLE
Remove inaccurate information

### PR DESCRIPTION
**Pull request by Youssef1313**
The article says:

> errors (also called exceptions) fall into one of three categories: syntax errors, run-time errors, and logic errors.

I don't think that referring to errors as exception is correct, since the syntax errors and logic errors aren't exceptions.

### CODE REVIEW

File: docs/visual-basic/programming-guide/language-features/error-types.md [hunk 1]
```diff
@@ -12,20 +12,20 @@ helpviewer_keywords:
...
 # Error Types (Visual Basic)
-In Visual Basic, errors (also called *exceptions*) fall into one of three categories: syntax errors, run-time errors, and logic errors.  
-  
-## Syntax Errors  
- *Syntax errors* are those that appear while you write code. Visual Basic checks your code as you type it in the **Code Editor** window and alerts you if you make a mistake, such as misspelling a word or using a language element improperly. Syntax errors are the most common type of errors. You can fix them easily in the coding environment as soon as they occur.  
-  
+In Visual Basic, errors fall into one of three categories: syntax errors, run-time errors, and logic errors.
+
+## Syntax Errors
+ *Syntax errors* are those that appear while you write code. Visual Basic checks your code as you type it in the **Code Editor** window and alerts you if you make a mistake, such as misspelling a word or using a language element improperly. Syntax errors are the most common type of errors. You can fix them easily in the coding environment as soon as they occur.
```
Comment (rpetrusha) [inline docs/visual-basic/programming-guide/language-features/error-types.md]:
```suggestion
 *Syntax errors* are those that appear while you write code. If you're using Visual Studio, Visual Basic checks your code as you type it in the **Code Editor** window and alerts you if you make a mistake, such as misspelling a word or using a language element improperly. If you compile from the command line, Visual Basic displays a compiler error with information about the syntax error. Syntax errors are the most common type of errors. You can fix them easily in the coding environment as soon as they occur.
```

File: docs/visual-basic/programming-guide/language-features/error-types.md [hunk 2]
```diff
@@ -12,20 +12,20 @@ helpviewer_keywords:
...
-  
-## Logic Errors  
- *Logic errors* are those that appear once the application is in use. They are most often unwanted or unexpected results in response to user actions. For example, a mistyped key or other outside influence might cause your application to stop working within expected parameters, or altogether. Logic errors are generally the hardest type to fix, since it is not always clear where they originate.  
+>  The `Option Explicit` statement is one means of avoiding syntax errors. It forces you to declare, in advance, all the variables to be used in the application. Therefore, when those variables are used in the code, any typographic errors are caught immediately and can be fixed.
+
+## Run-Time Errors
+ *Run-time errors* are those that appear only after you compile and run your code. These involve code that may appear to be correct in that it has no syntax errors, but that will not execute. For example, you might correctly write a line of code to open a file. But if the file is corrupted, the application cannot carry out the `Open` function, and it stops running. You can fix most run-time errors by rewriting the faulty code, and then recompiling and rerunning it.
```
Comment (rpetrusha) [inline docs/visual-basic/programming-guide/language-features/error-types.md]:
```suggestion
 *Run-time errors* are those that appear only after you compile and run your code. These involve code that may appear to be correct in that it has no syntax errors, but that will not execute. For example, you might correctly write a line of code to open a file. But if the file does not exist, the application cannot open the file, and it throws an exception. You can fix most run-time errors by rewriting the faulty code or by using [exception handling](../../language-reference/statements/try-catch-finally-statement.md), and then recompiling and rerunning it.
```

File: docs/visual-basic/programming-guide/language-features/error-types.md [hunk 3]
```diff
@@ -12,20 +12,20 @@ helpviewer_keywords:
...
   
+## Logic Errors
+ *Logic errors* are those that appear once the application is in use. They are most often unwanted or unexpected results in response to user actions. For example, a mistyped key or other outside influence might cause your application to stop working within expected parameters, or altogether. Logic errors are generally the hardest type to fix, since it is not always clear where they originate.
```
Comment (rpetrusha) [inline docs/visual-basic/programming-guide/language-features/error-types.md]:
```suggestion
 *Logic errors* are those that appear once the application is in use. They are most often faulty assumptions made by the developer, or unwanted or unexpected results in response to user actions. For example, a mistyped key might provide incorrect information to a method, or you may assume that a valid value is always supplied to a method that is not the case. Although logic errors can be handled by using [exception handling](../../language-reference/statements/try-catch-finally-statement.md) (for example, by testing whether an argument is `Nothing` and throwing a <xref:System.ArgumentNullException>), most commonly they should be addressed by correcting the error in logic and recompiling the application.
```

**Review by rpetrusha — CHANGES_REQUESTED** [review]
Thanks, @Youssef1313. You've identified a major weakness in a topic that is in serious need of revision. I've suggested some quick-and-dirty fixes to improve the topic a bit more. Let me know what you think.

**Comment by Youssef1313** [comment]
Thanks for your suggestions, @rpetrusha :)
I've applied them, I just replaced `a ArgumentNullException` with `an ArgumentNullException`.

**Review by rpetrusha — APPROVED** [review]

**Comment by rpetrusha** [comment]
Thanks for making the additional changes, @Youssef1313. I'll merge your PR as soon as our internal build completes successfully.

**Pull request merged**